File: src/main/java/Inversion.java
package main.java;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Inversion {


    public static void main(String[] args) throws IOException {

        Path path = Paths.get("src/java/resources/fileTest.txt");

        var numbers  = Files.readAllLines(path).stream().mapToInt(Integer::parseInt).toArray();



    }


    private static int[] mergeSort (int [] numbers ){

        if (numbers.length == 1){
            return numbers;
        }

        //Get the middle
        var middle = Math.floorDiv(numbers.length , 2);

         arrayLeft = Arrays.spliterator(numbers, 0, middle);

        var arrayRight = Arrays.spliterator(numbers, middle, numbers.length);







    }


    private static int[] merge (int [] left, int [] right){

        int [] result = new int[left.length + right.length];
        var indexResult = 0;
        var indexLeft = 0;
        var indexRight = 0;

        while (indexLeft < left.length && indexRight < right.length) {
            if (left[indexLeft] < right[indexRight]) {
                result[indexResult] = left[indexLeft];
                indexLeft++;
            } else {
                result[indexResult] = right[indexRight];
                indexRight++;
            }
            indexResult++;
        }

        return    Arrays.spliterator(left, indexLeft, left.length) + Arrays.spliterator(right, indexRight, right.length)
    }


}

File: inversion.py

import math

count_inversions = 0

def merge (array_left , array_right):
    result = []
    index_left = 0
    index_right = 0
    global count_inversions 

    while index_left < len(array_left) and index_right < len(array_right):
        if array_left[index_left] < array_right[index_right]:
            result.append(array_left[index_left])
            index_left+=1
        else :
            result.append(array_right[index_right])
            index_right+=1
            count_inversions+=len(array_left[index_left:])
    
    return result + array_left[index_left:] + array_right[index_right:]

def mergeSort (array):

    if len(array) == 1:
        return array
    
    middle = math.floor(len(array) / 2)

    array_left = array[:middle]
    array_right = array[middle:]

    return merge(mergeSort(array_left), mergeSort(array_right))



f=open("src/main/resources/input_final.txt", "r")
contents =f.readlines()
int_list = [int(i) for i in contents]
mergeSort(int_list)
print(count_inversions)